### PR TITLE
[fix]: [CDS-72312]: Not allowed keywords

### PR DESCRIPTION
**Pull request by vtxorxwitty**
## Describe your changes

FIX - Added keywords we want to restrict the names of variables same as what we do it from the UI. Appropriate error message is sent in the API response from the BE.

IMPACT - Customers won't be able to create variables with names mentioned in the screenshots above.

TESTING - Tested locally. Refer to the screenshot below. We can see the error message coming from the BE in the API response.

FF - No FF added/removed.

image

Screenshot 2023-06-22 at 3 33 22 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49463)

### CODE REVIEW

**Comment by vtxorxwitty** [comment]
trigger uts

**Comment by vtxorxwitty** [comment]
trigger uts

**Comment by vtxorxwitty** [comment]
trigger uts

File: 970-ng-commons/src/main/java/io/harness/validator/NGVariableNameValidator.java [hunk 1]
```diff
@@ -23,8 +23,9 @@ public class NGVariableNameValidator implements ConstraintValidator<NGVariableNa
   public static final String ALLOWED_CHARS_STRING_DEFAULT =
       "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_. ";
 
-  static final Set<String> NOT_ALLOWED_STRING_SET_DEFAULT = Sets.newHashSet("or", "and", "eq", "ne", "lt", "gt", "le",
-      "ge", "div", "mod", "not", "null", "true", "false", "new", "var", "return");
+  static final Set<String> NOT_ALLOWED_STRING_SET_DEFAULT =
```
Comment (ankittiwari-harness) [inline 970-ng-commons/src/main/java/io/harness/validator/NGVariableNameValidator.java:26]: nit: can we use one approach for set creation. Set.newHashSet or StreamCollect

**Comment by naidu-harness** [comment]
There will be more keywords we cannot support. can we check if there is a public OSS lib or file which contains all reserved keywords of JEXL and add them to avoid

**Comment by vtxorxwitty** [comment]
Sure @naidu-harness . Let me check if there is a library that checks on this.

**Comment by vtxorxwitty** [comment]
trigger uts

**Review by ankittiwari-harness — APPROVED** [review]

**Review by brijeshdhakar — APPROVED** [review]
> Reviewed 1 of 1 files at r1, 2 of 2 files at r2, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49463)* status: 4 file(s) to review.

**Review by Y0709 — APPROVED** [review]

**Comment by vtxorxwitty** [comment]
trigger review

**Pull request merged**